core/metadata.go

```diff
@@ -41,6 +41,12 @@ func (meta Metadata) Delete(key string) {
 	delete(meta, key)
 }
 
+// HasValue returns true if the given key exists
+func (meta Metadata) HasValue(key string) bool {
```

Comment:
Maybe we should make this "TryGetValue" as you normally directly want to do something once you know a key actually exists. Thoughts?